### PR TITLE
GCI-329: Implement Orders API authentication

### PR DESCRIPTION
**Pull request by VaughanJackson123**
* Authentication implemented according to the **Authentication** column of the two tables on this Confluence page: https://companieshouse.atlassian.net/wiki/spaces/CHS/pages/1515684399/GCI-926+Investigate+authentication+and+authorisation+checks+in+Orders+API.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/orders/api/util/TestConstants.java [hunk 1]
```diff
@@ -8,8 +8,10 @@
     public static final String ERIC_IDENTITY_HEADER_NAME = "ERIC-Identity";
     public static final String ERIC_IDENTITY_VALUE = "Y2VkZWVlMzhlZWFjY2M4MzQ3MT";
     public static final String ERIC_IDENTITY_TYPE_HEADER_NAME = "ERIC-Identity-Type";
-    public static final String ERIC_IDENTITY_TYPE_VALUE = "oauth2";
+    public static final String ERIC_IDENTITY_OAUTH2_TYPE_VALUE = "oauth2";
```
Comment (rjadhav-ch) [inline src/test/java/uk/gov/companieshouse/orders/api/util/TestConstants.java:11]: Could this perhaps be best named as `ERIC_IDENTITY_AUTH_TYPE_VALUE`?

Comment (VaughanJackson123) [reply src/test/java/uk/gov/companieshouse/orders/api/util/TestConstants.java:11]: Thanks for the suggestion. What is the reasoning? I think it does actually represent an OAuth2 identity?

Comment (rjadhav-ch) [reply src/test/java/uk/gov/companieshouse/orders/api/util/TestConstants.java:11]: I assumed it was a variable to describe Auth type, but it appears it's just describing OAuth2 :)

Comment (VaughanJackson123) [reply src/test/java/uk/gov/companieshouse/orders/api/util/TestConstants.java:11]: Yes, I think you are correct, it represents one possible authentication type, (that behind) OAuth2. Here, the other auth type is an API key.

**Pull request merged**